src/Symfony/Component/Translation/Provider/AbstractProvider.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Provider;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Translation\Exception\LogicException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ *
+ * @experimental in 5.3
+ */
+abstract class AbstractProvider implements ProviderInterface
```

Comment:
Here is a proposal:

Instead of this class, all providers should use relative URLs in their API calls.

And *factories* should be responsible for creating pre-configured http-client instances, by using `ScopingHttpClient`. This would make symfony/http-client a hard dep. Not an issue to me.

Reply:
See https://github.com/symfony/symfony/pull/40306, which should allow decoupling this concern from the component.